packages/server/lib/browsers/webdriver-classic/index.ts

```diff
@@ -0,0 +1,272 @@
+        browserName: string
+        browserVersion: string
+        platformName: string
+        pageLoadStrategy: 'normal'
+        strictFileInteractability: boolean
+        timeouts: {
+          implicit: number
+          pageLoad: number
+          script: number
+        }
+        'moz:accessibilityChecks': boolean
+        'moz:buildID': string
+        'moz:geckodriverVersion': string
+        'moz:debuggerAddress': string
+        'moz:headless': boolean
+        'moz:platformVersion': string
+        'moz:processID': number
+        'moz:profile': string
+        'moz:shutdownTimeout': number
+        'moz:webdriverClick': boolean
+        'moz:windowless': boolean
+        userAgent: string
+        sessionId: string
+      }
+    }
+  }
+}
+
+export class WebDriverClassic {
+  #host: string
+  #port: number
+  private sessionId: string | null = null
+
+  constructor (host: string, port: number) {
+    this.#host = host
+    this.#port = port
+  }
+
+  /**
+   * Creates a new WebDriver Session through GeckoDriver. Capabilities are predetermined
+   * @see https://www.w3.org/TR/webdriver2/#new-session
+   * @returns {Promise<WebDriver.Session.NewResult>} - the results of the Webdriver Session (enabled through remote.active-protocols)
+   */
+  async createSession (args: {
+    capabilities: {[key: string]: any}
+  }): Promise<WebDriver.Session.NewResult> {
+    const getSessionUrl = `http://${this.#host}:${this.#port}/session`
+
+    const headers = new Headers()
+
+    headers.append('Content-Type', 'application/json')
+
+    const body = {
+      capabilities: args.capabilities,
+    }
+
+    try {
+      const createSessionResp = await crossFetch(getSessionUrl, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(body),
+      })
+
+      if (!createSessionResp.ok) {
+        const error = new Error(`${createSessionResp.status}: ${createSessionResp.statusText}.`)
+
+        try {
+          const resp = await createSessionResp.json()
+
+          error.message = `${error.message } ${resp.value.error}. ${resp.value.message}.`
+        } finally {
+          // if for some reason we can't parse the response, continue to throw with some information.
+          throw error
+        }
+      }
+
+      const createSessionRespBody = await createSessionResp.json()
+
+      this.sessionId = createSessionRespBody.value.sessionId
+
+      return createSessionRespBody.value
+    } catch (e) {
+      debug(`unable to create new Webdriver session: ${e}`)
+      throw e
+    }
+  }
+
+  /**
+   * Gets available windows handles in the browser. The order in which the window handles are returned is arbitrary.
+   * @see https://www.w3.org/TR/webdriver2/#get-window-handles
+   *
+   * @returns {Promise<string[]>} All the available top-level contexts/handles
+   */
+  async getWindowHandles (): Promise<string[]> {
+    const getWindowHandles = `http://${this.#host}:${this.#port}/session/${this.sessionId}/window/handles`
```

Comment:
I don't think so anymore. I set it to null when ending the session but since we have to wait for that and we just kill the process, I removed the end session method. So this is just a string now. Fixed in 721100a489